src/actix/api/query_api.rs

```diff
@@ -110,13 +123,19 @@ async fn query_points_batch(
 
     let result = async move {
         let mut batch = Vec::with_capacity(searches.len());
-        for request in searches {
+        let mut inference_usages: Vec<Option<InferenceUsage>> = Vec::with_capacity(searches.len());
+
+        for request_item in searches {
             let QueryRequest {
                 internal,
                 shard_key,
-            } = request;
+            } = request_item;
+
+            let CollectionQueryRequestWithUsage { request, usage } =
+                convert_query_request_from_rest(internal, &inference_token).await?;
+
+            inference_usages.push(usage);
```

Comment:
Why are we collecting all usages, if we're only using only the last `InferenceUsage` item?